src/FirstSteps.Tests/RPG/Adventures/MineTests.cs

```diff
@@ -7,6 +7,35 @@ namespace FirstSteps.Tests.RPG.Adventures
 {
     public class MineTests
     {
-        
+        [Fact]
+
+        public void Dwarf_Should_Gain_One_Coin_After_the_First_Visit__In_The_Mine()
+        {
+            var mine = new Mine();
+            var dwarf = new Dwarf("Test");
+
+            mine.Enter(dwarf);
+
+            dwarf.AddCoins(1);
```

Comment:
Take a look one more time at the example:
```csharp
[Fact]
public void Undead_Should_Gain_One_Mana_Point_After_First_Visit_In_The_Dungeons()
{
    //arrange - section to setup the class you want to test
    var dungeons = new Dungeons();
    var undead = new Undead("Test");

    //act - section to perform the action you want to test
    dungeons.Enter(undead);

    //assert - section to check if the result matches your expectation
    undead.ManaPoints.Should().Be(1);
}
```
In the `assert` section, we check the number of mana points. So, instead of `dwarf.AddCoins(1);` you should check if the dwarf has as many coins as you expect after one visit to the mine. 
If you don't have the access to information about coins you can create public property and expose it.